subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java

```diff
@@ -130,6 +130,12 @@ private CredentialListener credentialListener() {
     }
 
     private boolean areCredentialsSafe(String identity, Credentials toCheck) {
+        // TODO:RC not using a provider does not necessarily imply unsafe credentials
+        // https://github.com/gradle/gradle/issues/22618
+        return true /* isUsingCredentialsProvider(identity, toCheck) */;
+    }
+
+    private boolean isUsingCredentialsProvider(String identity, Credentials toCheck) {
```

Comment:
🔴 If the function is no longer used we should remove it.

Reply:
We could take this all the way though, right? Including removing the CredentialListener interface, since it will never be called.

I figured the next step towards a proper solution could benefit from having this code as a starting point - the proper solution may basically rely on a new (user-implementable) provider type, and this code would work as is.

Reply:
> We could take this all the way though, right? Including removing the CredentialListener interface

Yes. Ideally the change would be a single commit that could be reverted if we decide to use the previous implementation as a starting point.